### PR TITLE
unpaywall: [WEB-123] Rearchitect to allow for existing prmSearchResultAvailabilityLineAfter component

### PR DESCRIPTION
**Pull request by elizoller**
This PR includes restructure to allow the unpaywall functionality to extend an already existing prmSearchResultAvailabilityLineAfter component, fix name collision of template and module, move icon so the hover works.
You don't have to merge it. I just had to make modifications in order for it to work for me because our view package contains multiple enhancements on the prmSearchResultAvailabilityLineAfter.
I modeled the re-architecture after what I saw here: https://github.com/UMNLibraries/primo-explore-hathitrust-availability.
Feel free to delete if it doesn't fit your use case.

### CODE REVIEW

File: packages/unpaywall/src/unpaywall.module.js [hunk 1]
```diff
@@ -1,117 +1,124 @@
...
       }
+
     }
   ])
-  .component('prmSearchResultAvailabilityLineAfter', {
-    bindings: { parentCtrl: '<' },
+  .component('bulibUnpaywall', {
+    require: {
+      prmSearchResultAvailabilityLine: '^prmSearchResultAvailabilityLine'
+    },
```
[the remark this answers is not in the document]
Comment (elizoller) [reply packages/unpaywall/src/unpaywall.module.js:115]: by doing the component in the style
```
.component('prmSearchResultAvailabilityLineAfter', {
    bindings: { parentCtrl: '<' },
}
```
it collides with any other component called prmSearchResultAvailabilityLineAfter.
but if you don't use prmSearchResultAvailabilityLineAfter for your component name then you don't get the parentCtrl with the binding so you need to use the require to get that controller.

File: packages/unpaywall/index.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+require('./src/unpaywall.module');
+module.exports = 'bulibUnpaywall';
```
[the remark this answers is not in the document]
Comment (elizoller) [reply packages/unpaywall/index.js]: tested this, and it appears unnecessary. i'll remove.

File: packages/unpaywall/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "primo-explore-unpaywall",
-  "version": "1.1.7",
+  "version": "1.1.8",
   "description": "Add 'Open Access available via unpaywall' link to search-result-avaliability-line-after in Primo New UI",
   "main": "src/unpaywall.module.js",
```
[the remark this answers is not in the document]
Comment (elizoller) [reply packages/unpaywall/package.json:5]: i believe this will become irrelevant then.

**Comment by atla5** [comment]
@elizoller i've published a new version at `1.2.0` with some basic docs changes. can you confirm that this works on your end and then i'll merge it?

**Comment by atla5** [comment]
we're now using this updated version ourselves, and it's working for us. so i'll count this as successful and merge it. let us know if you have any issues with it going forward.

and thanks again for the contribution!  🎉

**Pull request merged**